packages/result/src/lib/Option/IOption.ts

```diff
@@ -206,6 +223,23 @@ export interface IOption<T> {
 	 */
 	mapOrElse<U>(defaultValue: () => U, cb: (value: T) => U): U;
 
+	/**
+	 * Maps a `None` to the returned `Option<U>` by applying a function to a contained value, leaving `Some<T>`
+	 * untouched.
+	 * @param cb The predicate.
+	 *
+	 * @example
+	 * ```typescript
+	 * const maybeSomeString = some('Hello, world!');
+	 * const maybeSomeLength = maybeSomeString.map((value) => value.length);
+	 *
+	 * assert.equal(maybeSomeLength, some(13));
+	 * ```
+	 *
+	 * @note This is an extension not supported in Rust
+	 */
+	mapNoneInto<Inner>(cb: () => Option<Inner>): Option<T | Inner>;
```

Comment:
I don't understand why this returns a `Option<T | Inner>`. I thought it would just be a `Option<Inner>`.

Reply:
Because `Some<T>` will map to `Option<T>`, while `None` will map to `Option<Inner>`.

The classes have strict types.

Reply:
Wait, would same apply to `mapInto`? I don't think we care about the type of the initial `Some`, it's just about the new `Some`.